src/providers/conversion/coingecko/Coingecko.ts

```diff
@@ -0,0 +1,44 @@
+import fetch from 'cross-fetch';
+import { ConversionRateProvider, Currency } from '../ConversionRateProvider';
+
+export class Coingecko implements ConversionRateProvider {
+  constructor() {}
+
+  // this method translates currency strings to the format that coingecko requires
+  private static translateCurrency(currency: Currency): string {
+    switch (currency) {
+      case Currency.AR:
+        return 'arweave';
+      case Currency.SOL:
+        return 'solana';
+      case Currency.USD:
+        return 'usd';
+      case Currency.EUR:
+        return 'eur';
+      default:
+        throw new Error('Invalid currency supplied to Coingecko conversion rate provider');
```

Comment:
I guess we want to make this unified with the rest of the SDK errors inheriting that Exception class?

Reply:
Yep, u can define more errors. like `ERROR_COINGECKO_***`